skyportal/models/user_token.py

```diff
@@ -250,9 +261,11 @@ def delete_single_user_group(mapper, connection, target):
     single_user_group = target.single_user_group
 
     # Delete single-user group
-    @event.listens_for(DBSession(), "after_flush_postexec", once=True)
+    @event.listens_for(inspect(target).session, "after_flush_postexec", once=True)
     def receive_after_flush(session, context):
-        DBSession().delete(single_user_group)
+        if single_user_group:
+            new_single_user_group = session.merge(single_user_group)
+            session.delete(new_single_user_group)
```

Comment:
Its really very simple (took me only a few hours to figure out):
The `single_user_group` is saved along with the `receive_after_flush` function as part of a "closure". 
That group comes from a different session so it must be merged into the current session. 
The output of the merge must be assigned a different name, otherwise the closure doesn't work, because that name is used locally and overrides the name from external scope (it throws an undefined error for `single_user_group` even if that name is used after the external variable is used). 
Then it deletes that group and everyone is happy. 
Simple, right?

Reply:
Err, right ;) Please add that as an inline comment.